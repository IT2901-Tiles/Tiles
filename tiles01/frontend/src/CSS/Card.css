.card{
    width: 13vw;
    height: auto;
    margin: 4vw 2vw;
    border-radius: 5px;
}

.card:hover {
  box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
    0px 5px 22px 4px rgba(0, 0, 0, 0.12), 0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}

.card-active{
  transform: scale(1.25);
  box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
            0px 5px 22px 4px rgba(0, 0, 0, 0.12),
            0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}
  
/* STYLES for the category pages */
.chooseCardsButton {
    position: relative;
}

/** STYLES FOR ChosenCardsPage.jsx **/

/* Styles for the grids and wrappers */
.chosenCardsGrid {
    display: grid;
    grid-template-columns: auto auto auto auto;
    width: 100%;
    margin-top: 0.45vw;
    padding-top: 0.25vw;
}

.chosenCardsGrid2 {
    grid-area: 1 / 2 / 1 / 4;
    display: grid;
    grid-template-columns: auto auto auto; /* remove one "auto" if we decide to go with 4 cards instad of 3 */
    border-radius: 15px;
    padding-top: 3.5vw;
    padding-left: 3vw;
    padding-right: 3vw;
    padding-bottom: 3.5vw;
    background-color: #FBF4D7;
}

.wrapperCardNotChosen {
  height: 18.7vw; /* Must be 1.3vw smaller than chosenCardContainer smaller due to the border */
  position: relative;
  border: 5px;
  /* padding-bottom: 7%; must have slightly (about 4%) bigger padding than images chosen because of the border */
}

.chosenCardContainer,
#firstWrapperCardNotChosen {
  height: 20vw;
}

/** Styles for the cards **/

.chosenTrigger,
.chosenThings,
.chosenFeedback,
.noImgChosen {
  box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
    0px 5px 22px 4px rgba(0, 0, 0, 0.12), 0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}

.chosenTrigger:hover,
.chosenThings:hover,
.chosenFeedback:hover {
  opacity: 0.8;
}

.chosenTrigger,
.chosenThings,
.chosenFeedback,
.noImgChosen {
  height: 100%;
  border-radius: 5px; /* makes the edges of the cards/not picked cards rounded */
}

/** Styles when no card is picked **/
#noCardChosenTrigger,
#noCardChosenThings,
#noCardChosenFeedback {
  position: absolute;
  top: 15%;
  font-size: 2vw;
  transition-timing-function: ease-out;
  font-family: 'Comic Neue', sans-serif;
}

#noCardChosenTrigger {
  left: 36%;
}

#noCardChosenThings {
  left: 37%;
}

#noCardChosenFeedback {
  left: 32%;
}


#buttonChosenCardsPage {
  position: relative;
  left: 25%;
  width: 50%;
  top: -85px;
}

/** Responsive design for buttons and cards **/

/* Enlargens cards on Category page and ChosenCardspage*/
@media only screen and (max-width: 700px) {

    #buttonChosenCardsPage {
        top: -65%;
    }

    .card {
        width: 17vw;
    }

    .cardContainer {
      margin-top: 10vw;
    }

    .wrapperCardNotChosen {
        height: 23.7vw; /* Must be 1.3vw smaller than chosenCardContainer smaller due to the border */
    }
    
    .chosenCardContainer, #firstWrapperCardNotChosen {
        height: 24vw;
    }

    .chosenCardsGrid {
      padding-top: 0.5vw;
    }
}

@media only screen and (max-width: 600px) {
  #buttonChosenCardsPage {
    top: -60%;
  }
}

@media only screen and (max-width: 500px) {
  #buttonChosenCardsPage {
    top: -55%;
  }
}
