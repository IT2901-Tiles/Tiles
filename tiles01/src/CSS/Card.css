.card{
    width: 10rem;
    height: auto;
    margin: 5% 2%;
}

.card:hover {
    box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
            0px 5px 22px 4px rgba(0, 0, 0, 0.12),
            0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}
.card-active{
    transform: scale(1.3);
    box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
            0px 5px 22px 4px rgba(0, 0, 0, 0.12),
            0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}

/* STYLES FOR ChosenCardsPage.jsx */

/** Styles for the grids and wrappers **/
.chosenCardsGrid {
    display: grid;
    grid-template-columns: auto auto auto auto;
    width: 100%;
    margin-top: 2vh;
}

.chosenCardsGrid2 {
    grid-area: 1 / 2 / 1 / 4;
    display: grid;
    grid-template-columns: auto auto auto; /* remove one "auto" if we decide to go with 4 cards instad of 3 */
    border-radius: 20px; /* same border radius as other elements */
    padding: 3%;
    background-color: #FBF4D7;
}

.wrapperCardNotChosen {
    height: 18.7vw; /* Must be 1.3vw smaller than chosenCardContainer smaller due to the border */
    position: relative;
    /* padding-bottom: 7%; must have slightly (about 4%) bigger padding than images chosen because of the border */
}

.chosenCardContainer, #firstWrapperCardNotChosen {
    height: 20vw;
}

/** Styles for the cards **/
.chosenTrigger, .chosenThing, .chosenFeedback, .noImgChosen {
    box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
            0px 5px 22px 4px rgba(0, 0, 0, 0.12),
            0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}

.chosenTrigger:hover, .chosenThing:hover, .chosenFeedback:hover {
    opacity: 0.8;
}


.chosenTrigger, .chosenThing, .chosenFeedback, .noImgChosen {
    height: 100%;
}

/** Styles when no card is picked **/
#textNoCardChosen1, #textNoCardChosen2, #textNoCardChosen3 {
    position: absolute;
    top: 15%;
    font-size: 2vw;
    transition-timing-function: ease-out;
}

#textNoCardChosen1 { left: 36%; }

#textNoCardChosen2 { left: 39%; }

#textNoCardChosen3{ left: 32% }

#buttonChosenCardsPage {
    position: relative;
    left: 25%;
    width: 50%;
    top: -85px;
}

/** Responsive design for buttons **/

@media only screen and (max-width: 700px) {
    #buttonChosenCardsPage {
        top: -65%;
    }
}

@media only screen and (max-width: 600px) {
    #buttonChosenCardsPage {
        top: -60%;
    }
}

@media only screen and (max-width: 500px) {
    #buttonChosenCardsPage {
        top: -55%;
    }
}
