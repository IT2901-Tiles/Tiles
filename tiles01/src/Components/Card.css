.card{
    width:15%;
    margin: 3%;
}

.card:hover{
    box-shadow: 0px 12px 17px 2px rgba(0, 0, 0, 0.14),
            0px 5px 22px 4px rgba(0, 0, 0, 0.12),
            0px 7px 8px -4px rgba(0, 0, 0, 0.2);
}
.card-active{
    border: 4px blue solid;
}