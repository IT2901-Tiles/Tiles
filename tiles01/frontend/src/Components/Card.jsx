import React from 'react'
import "../CSS/Card.css"

// Card component which displays the image of one card
function Card(props) {
    function handleClick(event) {
        props.onClick(event.target)
    }
    let imageClass = "card"
    //check if this card is chosen
    if (parseInt(props.active) === parseInt(props.id)) {
        imageClass = "card card-active"
    }

    return (
        <img className={imageClass} name={props.name} onClick={handleClick} id={props.id} key={props.id} src={props.src} alt="A card" ></img>
    )
}

export default Card