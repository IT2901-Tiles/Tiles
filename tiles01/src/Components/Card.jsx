import React from 'react';
import "../CSS/Card.css";

function Card(props) {

    function handleClick(event){
        props.onClick(event.target) 
        console.log("Card click")
    }

    var imageClass="card"
    if (parseInt(props.active) === parseInt(props.id)){
        imageClass="card card-active"

    }

    return (
            <img className={imageClass} name={props.name} onClick={handleClick} id={props.id} key={props.id} src={props.src} alt="A card" ></img>
    );
}

export default Card;