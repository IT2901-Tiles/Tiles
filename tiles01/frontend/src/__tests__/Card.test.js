import {configure, shallow} from "enzyme"
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {cleanup} from '@testing-library/react';
import Card from '../Components/Card';
import renderer from 'react-test-renderer';


afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('Card', () => {
    const wrapper = shallow(<Card/>)
    it('renders correctly', () => {
        const tree = renderer.create(<Card/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('contains an image', () => {
        expect(wrapper.find('img')).toHaveLength(1);
    });
    it('has correct imageClass', () => {
        const initialProps = {
            active: "1",
            id: "1",
        }
        const container = shallow(<Card {...initialProps} />);
        expect(container.find('img').props().className).toEqual("card card-active");
    });

});