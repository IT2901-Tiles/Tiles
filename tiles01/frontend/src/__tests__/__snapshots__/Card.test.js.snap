// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<img
  alt="A card"
  className="card"
  onClick={[Function]}
/>
`;
